src/main.cpp

```diff
@@ -3324,13 +3324,13 @@ bool CBlock::ConnectBlock(CTxDB& txdb, CBlockIndex* pindex, bool fJustCheck, boo
     //  End of Network Consensus
 
     // Gridcoin: Track payments to CPID, and last block paid
-    if (IsResearcher(bb.cpid) && pindex->nHeight > nNewIndex2)
+    if (pindex->nResearchSubsidy > 0 && IsResearcher(bb.cpid))
```

Comment:
isnt nNewIndex2 when the ResearchSubsidy fields were added to blockchain? this won't cause an issue from before that point?

Reply:
It seems not. I wanted the runtime checks to be done the same way as when loading the chain and it does not use `nNewIndex2`.